warehouse/query-core/src/test/java/datawave/query/predicate/IsRootPointerPredicateTest.java

```diff
@@ -0,0 +1,66 @@
+package datawave.query.predicate;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Random;
+
+import org.apache.accumulo.core.data.Key;
+import org.junit.Test;
+
+import datawave.data.hash.HashUID;
+import datawave.data.hash.UID;
+
+public class IsRootPointerPredicateTest {
```

Comment:
Test name should match the class it is testing, i.e. "RootPointerPredicateTest". 

For extra clarity the tests could be broken out into more explicit names like 
testEventKeyWithContext
testTermFrequencyKeyWithContext
testFieldIndexKeyWithContext
...and then the same sans/without context

Please add a test that demonstrates the failure case for a uid that is too short.